tests/types/test_numpy.py

```diff
@@ -0,0 +1,131 @@
+import pytest
+from psycopg.adapt import PyFormat
+from psycopg.types.numpy import register_numpy
+
+pytest.importorskip("numpy")
+
+import numpy as np  # noqa: E402
+
+# np Numeric types
+#
+# np.int8 (-128, 127) -> smallint (alias int2)
+# np.int16 (-32768, 32767) -> smallint
+# np.int32 (-2147483648, 2147483647) -> integer (alias int4)
+# np.int64 (-9223372036854775808, 9223372036854775807) -> bigint (alias int8)
+# np.uint8 (0 - 255)
+
+register_numpy()
```

Comment:
The adapters shouldn't be registered globally: this pollutes the other tests.

Please register numpy types in the tests, on the connection only, or use a fixture to return a numpy-registered connection. See for instance:

https://github.com/psycopg/psycopg/blob/85c965089ae9cd7234dcced357b6e9b4a0637b6d/tests/types/test_shapely.py#L58-L69